tsrc/cli/__init__.py

```diff
@@ -25,9 +48,23 @@ def find_workspace_path():
     raise tsrc.Error("Could not find current workspace")
 
 
-def get_workspace(args):
-    if args.workspace_path:
-        workspace_path = path.Path(args.workspace_path)
+def workspace_cli(func):
+    @click.option("-w", "--workspace", metavar="<workspace>")
+    @click.pass_context
+    @main_wrapper
+    @functools.wraps(func)
+    def wrapped(ctx, *args, **kwargs):
+        ctx.obj = dict()
+        ctx.obj["workspace"] = get_workspace(kwargs["workspace"])
+        del kwargs["workspace"]
```

Comment:
Why do you delete the var?

Reply:
Because the workspace path should be retrieved from ctx.obj, and not a named argument.

If we don't delete this, code will crash with:

    TypeError: foo() got an unexpected keyword argument 'workspace'